.github/workflows/pr-slack-notification.yaml

```diff
@@ -0,0 +1,304 @@
+          "${{ env.API_URL }}?state=open&sort=created&direction=desc&per_page=100")
+          
+          # Load previous PR data if exists
+          if [ -f cache/notified_prs.json ]; then
+            previous_prs=$(cat cache/notified_prs.json)
+          else
+            previous_prs="[]"
+          fi
+
+          pr_list=""
+          new_notified_prs="[]"
+          notify=false
+
+          for pr_info in $(echo "$pr_infos" | jq -r '.[] | @base64'); do
+            _jq() {
+              echo "$pr_info" | base64 --decode | jq -r "${1}"
+            }
+
+            pr_number=$(_jq '.number')
+            pr_title=$(_jq '.title')
+            pr_user=$(_jq '.user.login')
+            pr_url=$(_jq '.html_url')
+            pr_draft=$(_jq '.draft')
+            pr_created_at=$(_jq '.created_at')
+            pr_updated_at=$(_jq '.updated_at')
+
+            pr_data=$(jq -n --arg number "$pr_number" --arg updated_at "$pr_updated_at" '{number: $number, updated_at: $updated_at}')
+            new_notified_prs=$(echo "$new_notified_prs" | jq --argjson pr_data "$pr_data" '. += [$pr_data]')
+
+            echo "New Notified PR List------ $new_notified_prs"
+
+            # Check if the PR is new or updated
+            previous_pr=$(echo "$previous_prs" | jq --arg number "$pr_number" '.[] | select(.number == $number)')
+            echo "Checking PR is new or not --- $previous_pr"
+            if [ -z "$previous_pr" ] || [ "$(echo "$previous_pr" | jq -r '.updated_at')" != "$pr_updated_at" ]; then
+              draft_status=""
+              echo "------Draft value ---- $pr_draft"
+              if [ "$pr_draft" = "true" ]; then
+                draft_status="\n*Draft*: true"
+              fi
+
+              pr_list="${pr_list}\n*Pull Request* #${pr_number}: ${pr_title}\n*Created by*: ${pr_user}\n*URL*: ${pr_url}${draft_status}\n*Created At*: ${pr_created_at}\n*Last Updated At*: ${pr_updated_at}\n"
+              notify=true
+              echo "Boolean Value for notify ------- $notify"
+            fi
+          done
+
+          # Save current PR data for future comparison
+          echo "$new_notified_prs" > cache/notified_prs.json
+
+          if [ "$notify" = true ]; then
+            echo -e "$pr_list" > pr_list.txt
+            echo "::set-output name=notify::true"
+          else
+            echo "::set-output name=notify::false"
+          fi
+
+      - name: Fetch closed pull requests since the last run
+        id: fetch_closed_prs
+        run: |
+          if [ -f cache/last_run_timestamp.txt ]; then
+            last_run_timestamp=$(cat cache/last_run_timestamp.txt)
+          else
+            last_run_timestamp=$(date -u -d "-24 hours" +%Y-%m-%dT%H:%M:%SZ)
+          fi
+
+          echo "Last run timestamp: $last_run_timestamp"
+
+          closed_prs=$(curl -s -H "Authorization: token ${{ secrets.GITHUB_TOKEN }}" \
+          "${{ env.API_URL }}?state=closed&sort=updated&direction=desc&per_page=100")
```

Comment:
why are we using this parameter "sort=updated"?

Reply:
Yes I modified the code.

Reply:
The GitHub API expects certain values for the state parameter, such as `open, closed, or all`, but it does not support combining states like `(closed || open)` directly.

If we want to fetch both open and closed pull requests, we can use `state=all` to include all pull requests regardless of their state. However, this approach has drawbacks. It includes all PRs (both open and closed) in the comparison with the `cached PR list`, which increases the time and computational requirements. Additionally, using `state=all` prevents us from specifying a limit like `per_page=100`, as it will fetch all PRs in the repository (currently around 384 PRs in lisp4ij).

<img width="751" alt="Screenshot 2024-06-26 at 10 23 57 AM" src="https://github.com/OpenLiberty/liberty-tools-intellij/assets/133869023/8afb5bbd-e842-4c4c-9364-b2eaaa0c9449">

Reply:
@vaisakhkannan I noticed you removed `sort=updated` without explaining what it was for. Was there a reason you had that parameter initially?

Reply:
@TrevCraw Initially, I thought it would be better to sort the closed PRs based on `sort=updated`. However, then I had realized that if we ran daily cron jobs, there was no need to sort, and the list may be within the 100 limit. If the number of closed PRs exceeds this limit, then we will definitely need `sort=updated`. So now I think based on Michael's comment aswell, I have added back to using `sort=updated`. In the future, we expect to have more than 100 closed PRs."

<img width="506" alt="Screenshot 2024-06-27 at 4 28 11 PM" src="https://github.com/OpenLiberty/liberty-tools-intellij/assets/133869023/453b9d1e-a96d-4bc2-a146-481da311040a">